### PR TITLE
Irrigation script

### PR DESCRIPTION
**Pull request by lpetrov**

### CODE REVIEW

File: SHELLY_MJS.md [hunk 1]
```diff
@@ -17,6 +17,13 @@ The script, when run, will fetch via REST api from a weather service the current
 cloud coverage is above or below certain percentage and respectively open or close window shades by calling a Shelly
 2.5 (Gen1) endpoint.
 
+precipitation-irrigation.js: Turn on/off watering based on precipitation in last 24 hours (based on AccuWeather data)
+===
+With this script you can stop or limit watering your garden according to the rain that has fallen in the last 24 hours.
```
Comment (Mircho) [inline SHELLY_MJS.md:22]: If there was precipitation in the past period (24h) skip an irrigation cycle. Data is retrieved from a public wather API service for the location.

File: precipitation-irrigation.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+// With this script you can stop or limit watering your garden according to the rain
+// that has fallen in the last 24 hours
+// You can use any Shelly Plus 1/Pro1 ot Pro2 to control your irrigation system
+// Dont forger to add AutoOFF for max Irrigation time and set a Schedule which start irrigation in device webUI.
+// Configure Accuweather APIKEY and end points
+let CONFIG = {
+    accuWeatherAPIKEY: "YourACCUWEATHERKeyGoesHere",
+//  weatherForecastEndpoint: "http://dataservice.accuweather.com/forecasts/v1/daily/1day/",
```
Comment (Mircho) [inline precipitation-irrigation.js:8]: remove commented line

File: precipitation-irrigation.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            print("RainValue", RainValue);
+            decideIfToIrrigate(RainValue);
+        }
+    );
+};
+
+// This function check are the rain in enought and switch off the valve
+function decideIfToIrrigate(RainValue) {
+    print(here, " Rain Last 24h - ", RainValue, " mm ");
+    if (RainValue > 4.5) {
```
Comment (Mircho) [inline precipitation-irrigation.js:51]: Move 4.5 to CONFIG

File: precipitation-irrigation.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+//        Can be use to caluculate irrigration time base of amount the Rain last 24h
+//        let seconds_to_irrigate = (10 - RainValue) * 30;
+//        Shelly.call("Switch.Set", {"id": 0, "on": true, "toggle_after": seconds_to_irrigate});
+        Shelly.call("Switch.Set", {"id": 0, "on": false}); // Disable if you calculate irrigration time
```
Comment (Mircho) [inline precipitation-irrigation.js:55]: Also which switch to operate should be in CONFIG

**Review by Mircho — COMMENTED** [review]
Small remarks, great addition to Shelly scripts

File: precipitation-irrigation.js (hunk 2 above)
Comment (lpetrov) [reply precipitation-irrigation.js]: Done

File: SHELLY_MJS.md (hunk 1 above)
Comment (lpetrov) [reply SHELLY_MJS.md]: Done

File: precipitation-irrigation.js (hunk 3 above)
Comment (lpetrov) [reply precipitation-irrigation.js]: Done

File: precipitation-irrigation.js (hunk 4 above)
Comment (lpetrov) [reply precipitation-irrigation.js]: Done

**Pull request merged**